src/app/api/user/route.ts

```diff
@@ -180,3 +187,10 @@ export const PUT = async (request: NextRequest) => {
     });
   }
 };
+
+export async function getUserByEmailServer(email: string) {
+  const user = await prisma.user.findUnique({
+    where: { email },
+  });
+  return user;
+}
```

Comment:
Next api route structure does not like this function in here. I see that the use case is to fetch the user to test their password. I would suggest you to move this function into a seperate file like `utils.ts` under this same folder.